### PR TITLE
add deployment and util test for new support bundle service

### PR DESCRIPTION
**Pull request by zbj6266**

### CODE REVIEW

File: api/v1alpha1/cdapmaster_types.go [hunk 1]
```diff
@@ -80,7 +80,7 @@ type CDAPMasterSpec struct {
 	// This is an optional service and may not be required for CDAP to be operational.
 	// To disable this service: either omit or set the field to nil
 	// To enable this service: set it to a pointer to a SupportBundleSpec struct (can be an empty struct)
-	SupportBundle *SupportBundleSpec `json:"supportBundle,omitempty"`
+	SupportBundle SupportBundleSpec `json:"supportBundle,omitempty"`
```
Comment (chtyim) [inline api/v1alpha1/cdapmaster_types.go]: Why removed the optional? The support bundle should be an optional service.

File: controllers/deployment_test.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ var _ = Describe("Controller Suite", func() {
 		diffJson := func(expected, actual []byte) {
 			opts := jsondiff.DefaultConsoleOptions()
 			diff, text := jsondiff.Compare(expected, actual, &opts)
+			print("printout actual: ", string(actual))
```
Comment (chtyim) [inline controllers/deployment_test.go]: Don't use print.

File: controllers/deployment.go [hunk 3]
```diff
@@ -43,12 +43,13 @@ func (d *DeploymentPlan) Init() {
...
-			"preview":   {servicePreview},
-			"appfabric": {serviceAppFabric},
-			"runtime":   {serviceRuntime},
+			"logs":          {serviceLogs},
+			"messaging":     {serviceMessaging},
+			"metrics":       {serviceMetrics},
+			"preview":       {servicePreview},
+			"appfabric":     {serviceAppFabric},
+			"runtime":       {serviceRuntime},
+			"supportbundle": {serviceSupportBundle},
```
Comment (chtyim) [inline controllers/deployment.go:52]: Why the support bundle service is a statefulset?

Comment (zbj6266) [reply controllers/deployment.go:52]: customer will store their support bundle in local dir for now so we need storage

File: README.md [hunk 4]
```diff
@@ -66,4 +66,4 @@ A step by step guide of running CDAP in Kubernetes using CDAP operator can be fo
 
 1. Install [kubebuilder](https://book-v1.book.kubebuilder.io/quick_start.html).
 
-2. Run `make test`
\ No newline at end of file
+2. Run `make test`
```
Comment (dli357) [inline README.md]: What exactly changed in this line?

File: config/crd/bases/cdap.cdap.io_cdapmasters.yaml [hunk 5]
```diff
@@ -1,4 +1,3 @@
-
```
Comment (dli357) [inline config/crd/bases/cdap.cdap.io_cdapmasters.yaml:1]: Were these removals intentional or auto-format?

File: controllers/testdata/supportbundle.json [hunk 6]
```diff
@@ -0,0 +1,279 @@
+{
+  "apiVersion":"apps/v1",
+  "kind":"StatefulSet",
+  "metadata":{
+    "annotations": {
+      "deployment.kubernetes.io/revision": "1"
+    },
+    "name":"cdap-test-supportbundle",
+    "namespace":"default",
+    "creationTimestamp":null,
```
Comment (dli357) [inline controllers/testdata/supportbundle.json:10]: FYI for expected output, please ensure only the expected output fields (and no extraneous) fields are included in the test template data, as there will be a fix to CDAP operator which changes the integration test JSON diff comparison behavior (see [CDAP-18732](https://cdap.atlassian.net/browse/CDAP-18732) for additional info).

**Review by dli357 — APPROVED** [review]
Other than some minor questions, LGTM. Thanks!

File: README.md (hunk 4 above)
Comment (zbj6266) [reply README.md]: Not sure I will rebase it

File: config/crd/bases/cdap.cdap.io_cdapmasters.yaml (hunk 5 above)
Comment (zbj6266) [reply config/crd/bases/cdap.cdap.io_cdapmasters.yaml:1]: it's intentional removal

File: controllers/testdata/supportbundle.json (hunk 6 above)
Comment (zbj6266) [reply controllers/testdata/supportbundle.json:10]: yeah I got the test result from the expected output

**Pull request merged**